Model/ConfigProvider/Method/Creditcard.php

```diff
@@ -124,39 +127,38 @@ class Creditcard extends AbstractConfigProvider
      */
     public function formatIssuers()
     {
-        $issuers = parent::formatIssuers();
-        $allowed = explode(',', $this->scopeConfig->getValue(
-            self::XPATH_CREDITCARD_ALLOWED_CREDITCARDS,
-            \Magento\Store\Model\ScopeInterface::SCOPE_STORE)
-        );
-
         $sorted = explode(',', $this->scopeConfig->getValue(
-            self::XPATH_CREDITCARD_SORT,
+            static::XPATH_CREDITCARD_SORT,
             \Magento\Store\Model\ScopeInterface::SCOPE_STORE)
         );
 
         if (!empty($sorted)) {
             $sortedPosition = 1;
             foreach ($sorted as $cardName) {
-                foreach ($issuers as $key => $issuer) {
-                    if ($issuers[$key]['name'] == $cardName && empty($issuers[$key]['sort'])) {
-                        $issuers[$key]['sort'] = $sortedPosition;
-                        $sortedPosition++;
-                        break;
-                    }
-                }
+                $sorted_array[$cardName] = $sortedPosition++;
             }
         }
 
-        usort($issuers, function ($cardA, $cardB){
-            return $cardA['sort'] - $cardB['sort'];
-        });
+        $issuers = parent::formatIssuers();
```

Comment:
Can we add some comments to these 20(?) lines on whats going on here? Just to make the mental load a bit lighter. And/or maybe move this to separate methods with descriptive names